screenpipe-audio/src/whisper/process_chunk.rs

```diff
@@ -1,62 +1,118 @@
 use std::collections::HashSet;
+use tracing::info;
+use whisper_rs::{get_lang_str, FullParams, SamplingStrategy, WhisperContext, WhisperState};
 
 lazy_static! {
     static ref TOKEN_REGEX: Regex = Regex::new(r"<\|\d{1,2}\.\d{1,2}\|>").unwrap();
 }
 
 pub fn process_with_whisper(
-    whisper_model: &mut WhisperModel,
+    ctx: &WhisperContext,
     audio: &[f32],
-    mel_filters: &[f32],
+    // mel_filters: &[f32],
     languages: Vec<Language>,
 ) -> Result<String> {
-    let model = &mut whisper_model.model;
-    let tokenizer = &whisper_model.tokenizer;
-    let device = &whisper_model.device;
-
-    debug!("converting pcm to mel spectrogram");
-    let mel = audio::pcm_to_mel(model.config(), audio, mel_filters);
-    let mel_len = mel.len();
-
-    debug!("creating tensor from mel spectrogram");
-    let mel = Tensor::from_vec(
-        mel,
-        (
-            1,
-            model.config().num_mel_bins,
-            mel_len / model.config().num_mel_bins,
-        ),
-        device,
-    )?;
-
-    debug!("detecting language");
-    let language_token = Some(multilingual::detect_language(
-        model,
-        tokenizer,
-        &mel,
-        languages.clone(),
-    )?);
-
-    debug!("initializing decoder");
-    let mut dc = Decoder::new(model, tokenizer, 42, device, language_token, true, false)?;
-
-    debug!("starting decoding process");
-    let segments = dc.run(&mel)?;
-    debug!("decoding complete");
-
-    process_segments(segments)
+    // let model = &mut whisper_model.model;
+    // let tokenizer = &whisper_model.tokenizer;
+    // let device = &whisper_model.device;
+
+    // debug!("converting pcm to mel spectrogram");
+    // let mel = audio::pcm_to_mel(model.config(), audio, mel_filters);
+    // let mel_len = mel.len();
+
+    // debug!("creating tensor from mel spectrogram");
+    // let mel = Tensor::from_vec(
+    //     mel,
+    //     (
+    //         1,
+    //         model.config().num_mel_bins,
+    //         mel_len / model.config().num_mel_bins,
+    //     ),
+    //     device,
+    // )?;
+
+    // debug!("detecting language");
+    // let language_token = Some(multilingual::detect_language(
+    //     model,
+    //     tokenizer,
+    //     &mel,
+    //     languages.clone(),
+    // )?);
+
+    // debug!("initializing decoder");
+    // let mut dc = Decoder::new(model, tokenizer, 42, device, language_token, true, false)?;
+
+    // debug!("starting decoding process");
+    // let segments = dc.run(&mel)?;
+    // debug!("decoding complete");
+    let mut whisper_model = ctx.create_state().expect("failed to create key");
+
+    let mut params = FullParams::new(SamplingStrategy::Greedy { best_of: 0 });
+
+    // Edit params as needed.
+    // Set the number of threads to use to 1.
+    params.set_n_threads(1);
+    // Enable translation.
+    params.set_translate(true);
+    // Set the language to translate to to English.
+    params.set_language(Some("en"));
```

Comment:
multi language?

Reply:
I need to remove this line. below it is the language setting. this was quick and dirty